usl_pipeline/usl_pipeline/shared/entities.py

```diff
@@ -0,0 +1,36 @@
+from dataclasses import dataclass
+
+import numpy as np
+import numpy.typing as npt
+from rasterio import CRS
+
+
+@dataclass
+class ElevationHeader:
+    """Elevation header data class keeping information needed to convert
+    geo-spacial coordinates to raster cells and back.
+
+    Args:
+        col_count: Number of columns in a raster cell matrix.
+        row_count: Number of rows in a raster cell matrix.
+        x_ll_corner: X-coordinate of lower-left corner of a raster region.
+        y_ll_corner: Y-coordinate of lower-left corner of a raster region.
+        cell_size: Size of each raster cell.
+        nodata_value: Special data value that indicates that cells having this
+            value correspond to missing data.
+        crs: Optional value defining Coordinate Reference System.
+    """
+
+    col_count: int
+    row_count: int
+    x_ll_corner: float
+    y_ll_corner: float
+    cell_size: float
+    nodata_value: float
+    crs: CRS = None
+
+
+@dataclass
+class Elevation:
+    header: ElevationHeader
+    data: npt.NDArray[np.float64] = None
```

Comment:
We should add some type checking to the build to catch things like this.
The type here should be `npt.NDArray[np.float64] | None` or equivalently `Optional[npt.NDArray[np.float64]]` since it can be `None`
the type checking would also catch that we're setting this attribute to a list with the `tolist` call above.

Reply:
Done